crates/bevy_text/src/text2d.rs

```diff
@@ -86,7 +86,7 @@ pub fn extract_text2d_sprite(
                 HorizontalAlign::Right => Vec3::new(-width, 0.0, 0.0),
             };
 
-            let mut text_transform = *transform;
+            let mut text_transform = Transform::from(*transform);
```

Comment:
This looks surprising to read.

Reply:
Short term I think this is still the best approach. We need to extract + modify the scale, but we also still need transform multiplication ops. We should discuss how acceptable this pattern is generally (see my comment below), but idk if we should block on that.

Reply:
I've eliminated the confusing code, now it mirrors similar code in `bevy_ui::render::extract_text_uinodes` and avoids the conversion to `Transform`.